geps/gep-1619/index.md

```diff
@@ -736,52 +669,139 @@ spec:
 supported by some implementations due to their current designs.
 
+### Conformance Details
+
+TODO
+
 ### Open Questions
 
-- What happens when session persistence is broken because the backend is not up or healthy? If that's an error case, how should that be handled? Should the API dictate the http error code? Or should the API dictate fall back behavior?
 - What happens when session persistence causes traffic splitting scenarios to overload a backend?
 - Should we add status somewhere when a user gets in to a "risky" configuration with session persistence?
 - Should there be an API configuration field that specifies how already established sessions are handled?
+- How do implementations drain established sessions during backend upgrades without disruption?
+    - Do we need a "session draining timeout" as documented by [A55: xDS-Based Stateful Session Affinity for Proxyless gRPC](https://github.com/grpc/proposal/blob/master/A55-xds-stateful-session-affinity.md#background)
+      defined in this API?
 
 ## TODO
-The following are items that we intend to resolve before we consider this GEP implementable:
 
-- We need to identify and document requirements regarding session draining and migration. How do implementations drain established sessions during backend upgrades without disruption?
-  - Do we need a "session draining timeout" as documented by [A55: xDS-Based Stateful Session Affinity for Proxyless gRPC](https://github.com/grpc/proposal/blob/master/A55-xds-stateful-session-affinity.md#background)
-    defined in this API?
+The following are items that we intend to resolve in future revisions:
+
+- We need to identify and document requirements regarding session draining and migration.
 - We need to document sessions with Java in greater detail. Java standardized the API and behavior of session persistence long ago and would be worth examining.
 - We need to add a small section on compliance regarding the browser and client relationship.
 - We need to finish enumerating all the edge cases in [Expected API Behavior](#expected-api-behavior) and identify
 potential scenarios where session persistence could break so an implementation can implement session persistence in a
 predicable way.
+- We need to clean up the [Implementations](#implementations) table to make it more organized and readable.
 
 ## Alternatives
 
-### Alternate Session Persistence API
+### Timeout Config Alternative
+
+Alternatively, timeouts such as `AbsoluteTimeoutSeconds` and `IdleTimeoutSeconds` could be added to the API. This would
+allow users to control the duration of their sessions. However, as described in [Expires / Max-Age](#expires-max-age),
+we currently prescribe that if cookies are used, then all cookies MUST be session cookies. If timeout configuration is
+added, then we also need to add an API to distinguish between session cookies and permanent cookies. This distinction is
```

Comment:
Although historically I've been very much in favor of starting with an MVP for APIs, we're now starting to run into the pain of each relatively small addition to an API requiring yet another GEP. So if it's likely that we'll want to add something in the future, I think I'd rather just add it to start with and skip all the extra future work this would be to add as a separate GEP.

Reply:
Added it back in. I took a stab an API design that has cookie-specific configuration (everything thus far has been generic to sessions). I used a struct called `CookieConfig` to organize the new `LifetimeType` enum field (`Session` or `Permanent`).

I'm new to API design and just tried to copy exist GWAPI patterns as best I could and naming is hard, so please, chime in.

There is a bit of an awkward situation now that if `cookieConfig.lifetimeType` = `Permanent` then `absoluteTimeoutSeconds` needs to be set (we need a value for cookie attribute `Expire` or `Max-Age`). I feel like it's either:
1. A smell for poor API design
2. Or it's acceptable, but CEL can take care of this situation by enforcing a value in `absoluteTimeoutSeconds` when `cookieConfig.lifetimeType` = `Permanent`